pwndbg/gdblib/disasm/aarch64.py

```diff
@@ -187,9 +206,31 @@ def _set_annotation_string(self, instruction: PwndbgInstruction, emu: Emulator)
                 AARCH64_SINGLE_STORE_INSTRUCTIONS[instruction.id],
                 instruction.operands[1].str,
             )
+        elif instruction.id in AARCH64_MATH_INSTRUCTIONS:
+            self._common_binary_op_annotator(
+                instruction,
+                emu,
+                instruction.operands[0],
+                instruction.operands[-2].before_value,
+                instruction.operands[-1].before_value,
+                AARCH64_MATH_INSTRUCTIONS[instruction.id],
+            )
         else:
             self.annotation_handlers.get(instruction.id, lambda *a: None)(instruction, emu)
 
+    def _handle_adrp(self, instruction: PwndbgInstruction, emu: Emulator) -> None:
+        result_operand, right = instruction.operands
+        if result_operand.str and right.before_value is not None:
+            address = right.before_value
+
+            TELESCOPE_DEPTH = max(0, int(pwndbg.config.disasm_telescope_depth))
```

Comment:
Can `pwndbg.config.disasm_telescope_depth` ever be negative? Does it make any sense to set it negative instead of 0?

Reply:
If it should always be >=0 its parameter type should be set accordingly (`ZUINTEGER` or sth).

Reply:
But that's for another PR, please create an issue if that's the case here.